extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerWorking.java

```diff
@@ -11,38 +11,35 @@
 
 import org.jboss.logging.Logger;
 
-import io.quarkus.container.image.deployment.ContainerImageConfig;
 import io.quarkus.deployment.util.ExecUtil;
 
-public class DockerBuild implements BooleanSupplier {
+public class DockerWorking implements BooleanSupplier {
 
-    private static final Logger LOGGER = Logger.getLogger(DockerBuild.class.getName());
-    private static boolean daemonFound = false;
-
-    private final ContainerImageConfig containerImageConfig;
-
-    public DockerBuild(ContainerImageConfig containerImageConfig) {
-        this.containerImageConfig = containerImageConfig;
-    }
+    private static final Logger LOGGER = Logger.getLogger(DockerWorking.class.getName());
 
     @Override
     public boolean getAsBoolean() {
-        // No need to perform the check multiple times.
-        if (daemonFound) {
-            return true;
+        try {
+            if (!ExecUtil.execSilent("docker", "help")) {
```

Comment:
This returns a single line and can be faster than reading the whole output of `docker help`
```suggestion
            if (!ExecUtil.execSilent("docker", "-v")) {
```

Reply:
Good suggestion! I was looking for something like that and I totally missed it!

Reply:
Fixed